cloudinit/sources/__init__.py

```diff
@@ -1185,9 +1185,9 @@ def parse_cmdline() -> str:
     Passing by command line overrides runtime datasource detection
     """
     cmdline = util.get_cmdline()
-    ds_parse_0 = re.search(r"ds=([^\s;]+)", cmdline)
-    ds_parse_1 = re.search(r"ci\.ds=([^\s;]+)", cmdline)
-    ds_parse_2 = re.search(r"ci\.datasource=([^\s;]+)", cmdline)
+    ds_parse_0 = re.search(r"(?:^|\s)ds=([^\s;]+)", cmdline)
```

Comment:
The `?:` is meaningless in these regexes, did you intend something else?

Reply:
It makes the group non-capturing. The other group in these regexes is referred to as `.group(1)`, so if you capture the first group, you have to shift the group index. As the first group is only used to match what we want, I think making it non-capturing is "more correct" than shifting the group at the end of this function.